code/game/objects/items/devices/blood_scanner.dm

```diff
@@ -0,0 +1,92 @@
+		if(poked_guy.blood_volume <= BLOOD_VOLUME_SAFE)
+			render_list += "<span class='danger ml-1'>Warning: Dangerously low blood flow.</span>\n"
+		if(tox_loss > 10)
+			render_list += "<span class='danger ml-1'>Warning: Toxic buildup detected in bloodstream.</span>\n"
+		if(poked_guy.has_status_effect(/datum/status_effect/eigenstasium))
+			render_list += "<span class='danger ml-1'>Warning: Dimensional instability detected. Administer stabilizers.</span>\n"
+		if(poked_guy.has_reagent(/datum/reagent/gold/cursed) || poked_guy.has_status_effect(/datum/status_effect/midas_blight))
+			render_list += "<span class='danger ml-1'>Warning: Hemo-aurificating hexes present.</span>\n" //can it detect normal things? barely. But ancient greed-cursed magicks? Spot on.
+		if(HAS_TRAIT_FROM(poked_guy, TRAIT_NODEATH, /datum/reagent/inverse/penthrite))
+			render_list += "<span class='danger ml-1'>Warning: Frankensteinian revivification in progress.</span>\n" //patient is currently an immortal drug-zombie
+		if(poked_guy.has_status_effect(/datum/status_effect/high_blood_pressure))
+			render_list += "<span class='danger ml-1'>Warning: Dangerously high blood pressure.</span>\n"
+		if(HAS_TRAIT(poked_guy, TRAIT_IMMUNODEFICIENCY))
+			render_list += "<span class='danger ml-1'>Warning: Low immune cell concentration.</span>\n" //it doe
+		to_chat(user, boxed_message(jointext(render_list, "")), type = MESSAGE_TYPE_INFO)
+
+	else
+		if(prob(25))
+			poked_bit?.force_wound_upwards(/datum/wound/pierce/bleed/moderate/needle_fail, wound_source = "idiot moved with a needle in them")
+			user.visible_message(span_warning("[src]'s needle is ripped out, tearing a hole in [poked_guy]'s [poked_bit]!"), span_warning("Damnit! The needle is torn out, making a tiny hole in [poked_guy.p_their()] [poked_bit]"), ignored_mobs = poked_guy)
+			to_chat(poked_guy, span_userdanger("<b>OWWW!</b> The needle of [src] is ripped out, tearing a small hole in your [poked_bit]!"))
+		return
+
+	return ITEM_INTERACT_SUCCESS
+
+/obj/item/blood_scanner/interact_with_atom_secondary(atom/interacting_with, mob/living/user, list/modifiers)
+	if(!isliving(interacting_with))
+		return NONE
+	if(!istype(interacting_with, /mob/living/carbon))
+		to_chat(user, span_warning("The [src] doesn't work on that kind of life."))
+		return NONE
+	var/mob/living/carbon/poked_guy = interacting_with
+	if(!user.can_read(src) || user.is_blind())
+		to_chat(user, span_warning("You are unable to read the [src]'s screen!"))
+		return
+	var/obj/item/bodypart/poked_bit = poked_guy.get_bodypart(check_zone(user.zone_selected))
+	user.visible_message(span_notice("[user] pricks [poked_guy] with [src]."), span_notice("You begin scanning [poked_guy] with [src]."), ignored_mobs = poked_guy)
+	to_chat(poked_guy, span_notice("[user] slides the needle of [src] into your [poked_bit] and begins pressing buttons."))
+	var/success = do_after(user, 2 SECONDS, poked_guy)
```

Comment:
This code is duplicated (also in interact_with_atom). You should put this bit in a seperate proc, and call that in interact_with_atom and interact_with_atom_secondary